doc/man1/openssl-cmp.pod.in

```diff
@@ -0,0 +1,1142 @@
+
+=back
+
+
+=head2 Message transfer options
+
+=over 4
+
+=item B<-server> I<[http[s]://]address[:port]>
+
+The IP address or DNS hostname and optionally port (defaulting to 80 or 443)
+of the CMP server to connect to using HTTP(S) transport.
+
+=item B<-proxy> I<[http://]address[:port][/path]>
+
+The HTTP(S) proxy server to use for reaching the CMP server unless B<no_proxy>
+applies, see below.
+The optional "http://" prefix and any trailing path are ignored.
+Defaults to the environment variable C<http_proxy> if set, else C<HTTP_PROXY>
+in case no TLS is used, otherwise C<https_proxy> if set, else C<HTTPS_PROXY>.
+
+=item B<-no_proxy> I<addresses>
+List of IP addresses and/or DNS names of servers not use an HTTP(S) proxy for,
+separated by commas and/or whitespace.
+Default is from the environment variable C<no_proxy> if set, else C<NO_PROXY>.
+
+=item B<-path> I<remote_path>
+
+HTTP path at the CMP server (aka CMP alias) to use for POST requests.
+Defaults to "/".
+
+=item B<-msg_timeout> I<seconds>
+
+Number of seconds (or 0 for infinite) a CMP request-response message round trip
+is allowed to take before a timeout error is returned.
+Default is 120.
+
+=item B<-total_timeout> I<seconds>
+
+Maximum number seconds an overall enrollment transaction may take,
+including attempts polling for certificates on C<waiting> PKIStatus.
+Default is 0 (infinite).
+
+=back
+
+
+=head2 Server authentication options
+
+=over 4
+
+=item B<-trusted> I<filenames>
+
+When verifying signature-based protection of CMP response messages,
+these are the CA certificate(s) to trust while checking certificate chains
+during CMP server authentication.
+This option gives more flexibility than the B<-srvcert> option because
+it does not pin down the expected CMP server by allowing only one certificate.
+
+Multiple filenames may be given, separated by commas and/or whitespace.
+Each source may contain multiple certificates.
+
+=item B<-untrusted> I<sources>
+
+Non-trusted intermediate certificate(s) that may be useful
+for building certificate chains when verifying
+the CMP server (when checking signature-based CMP message protection),
+the own TLS client cert (when constructing the TLS client cert chain),
+and/or the newly enrolled certificate.
+These may get added to the extraCerts field sent in requests as far as needed.
+
+Multiple filenames may be given, separated by commas and/or whitespace.
+Each file may contain multiple certificates.
+
+=item B<-srvcert> I<filename>
+
+The specific CMP server certificate to use and directly trust (even if it is
+expired) when verifying signature-based protection of CMP response messages.
+May be set alternatively to the B<-trusted> option
+if the certificate is available and only this one shall be accepted.
+
+If set, the issuer of the certificate is also used as the recipient of the CMP
+request and as the expected sender of the CMP response,
+overriding any potential B<-recipient> option.
```

Comment:
I wonder why we need this option?

Reply:
I've re-phrased the section to make more clear when the -recipient option should be used.